vms/proposervm/block.go

```diff
@@ -218,10 +222,23 @@ func (p *postForkCommonComponents) buildChild(
 		return nil, err
 	}
 
+	var childBlockVrfSig []byte
+	// do we have the VRFSig activated ? if so, figure out the child block vrf signature.
+	if p.vm.Upgrades.IsEtnaActivated(newTimestamp) {
+		if shouldBuildSignedBlock {
+			childBlockVrfSig = block.NextBlockVRFSig(parentBlockVRFSig, p.vm.Config.StakingBLSKey, p.vm.ctx.ChainID, p.vm.ctx.NetworkID)
```

Comment:
I see. I believe that my code already handles the validation part correctly ( since it gets the BLS public key from the `ValidatorState.GetValidatorSet`).
For the block building, would it work if I were to retrieve the validator set from `ValidatorState.GetValidatorSet` and compare the public registered BLS key there with the block builder's public key ?